Added microphone ducking documentation (#247)
mixxxdj/manual: source/chapters/microphones.rst

DJ controllers typically do not supply phantom power required for condenser
microphones. Stand-alone audio interfaces typically do supply phantom power.

Microphone Ducking
------------------

> While using Mixxx, you might want to talk through your microphone as the music plays, therefore, you might need the music to automatically fade into the background each time you speak, or you might prefer lowering the volume of the music to a specific level. Microphone ducking helps you achieve this.

Reviewer: The word "might" appears a lot here.  We can simplify:

"While using Mixxx with a microphone, you might want the music to automatically fade into the background each time you speak, or you might prefer lowering the volume of the music to a specific level. Microphone ducking helps you achieve this. The music will 'duck' out of the way of the microphone audio."

Reviewer: (I think it's worth explaining the word "duck" just in case people are having trouble with that)

Author: Thank you. I agree. 
I have added a definition for ducking just before this paragraph on Line 417. You can read through and let me know if where it might need correction....

.. figure:: ../_static/microphone_ducking.png
   :align: center
   :width: 40%
   :figwidth: 90%
   :alt: The microphone ducking section
   :figclass: pretty-figures

   The microphone ducking controls

The Strength knob adjusts the microphone talkover ducking strength and its behaviour depends on any of the Microphone talkover modes described below.
Clicking  on the :guilabel:`Auto` button will switch it to any of three modes:

   * Off: When you talk through your microphone, the music  volume does not reduce.

>    * Auto: Talking through your microphone will automatically reduce the music volume. You can adjust the amount to which the volume is automatically reduced with the Strength knob.

Reviewer: "When you speak through the microphone, the music volume will be reduced depending on how loud you speak. You can adjust the amount by which the volume is automatically reduced with the Strength knob."

Author: I have made the change. It sounds better :)

>    * Manual: This sets a fixed level to which the music volume reduces to, regardless of the volume of the microphone input. When you talk through your microphone, you can manually set the amount to which the music volume is reduced by turning the Strength knob to a level that suits you.

Reviewer: I forget if the knob represents the volume is reduced *to*, or the amount the volume is reduced *by*.  I'm also not sure if those are identical things :).

Reviewer: "Manual: This sets a fixed level to which the music volume reduces to when the Talk button is on, regardless of the volume of the microphone input. When you talk through your microphone, you can manually set the amount to which the music volume is reduced by turning the Strength knob to a level that suits you."

I think that correctly describes how ducking works?  I honestly don't remember

Author: I have edited Manual mode to differentiate it from the description. The new description teaches the user how to adjust the music volume when he talks, using the Strength knob in manual mode.

DJ controllers typically do not supply phantom power required for condenser
microphones. Stand-alone audio interfaces typically do supply phantom power.

Microphone Ducking
------------------

> Microphone ducking is that effect you hear when you speak through the microphone, and the music volume is reduced by the presence of the voice signal coming through the microphone.

Reviewer 2: ```suggestion
Microphone ducking is a technique to improve the audibility of speech by reducing the volume of background music while a voice signal is recorded by the microphone and allows listeners to hear the voice more clearly and consistently.
```

While using Mixxx with a microphone, you might want the music to automatically fade into the background each time you speak, or you might prefer lowering the volume of the music to a specific level. Microphone ducking helps you achieve this.  The music will 'duck' out of the way of the microphone audio.

> .. figure:: ../_static/microphone_ducking.png

Reviewer 2: Can you put the screenshot in a `2.2` subdirectory and use the same naming scheme (dashes vs. underscores) like you do in #154?

Author: Okay. I will.

   :align: center
   :width: 40%
   :figwidth: 90%
   :alt: The microphone ducking section
   :figclass: pretty-figures

   The microphone ducking controls

> The Strength knob adjusts the microphone talkover ducking strength and its behaviour depends on any of the Microphone talkover modes described below.

Reviewer 2: ```suggestion
The :guilabel:`Strength` knob adjusts the microphone talkover ducking strength and its behaviour depends on any of the Microphone talkover modes described below.
```

Reviewer: @deborahtrez , you can incorporate jan's changes by selecting the "commit suggestion" button, or adding it to a batch.

Author: Okay. I will start using this method. Thank you Owen.

Clicking  on the :guilabel:`Auto` button will switch it to any of three modes:

>    * Off: When you speak through the microphone, the music  volume does not reduce.
>    * Auto: When you speak through the microphone, the music volume will be reduced when Mixxx detects your voice. You can adjust the amount by which the volume is automatically reduced with the Strength knob.
>    * Manual: In this mode, you have to remember to manually adjust the Strength knob to a desired level each time you speak. Otherwise, the music will play at the volume set by the Strength knob, regardless of the volume coming through the microphone input. When you speak through your microphone, set the volume to which the music should reduce by turning the Strength knob to a level that suits you.

Reviewer 2: ```suggestion
   * :guilabel:`Off`: When you speak through the microphone, the music  volume does not reduce.
   * :guilabel:`Auto`: When you speak through the microphone, the music volume will be reduced when Mixxx detects your voice. You can adjust the amount by which the volume is automatically reduced with the Strength knob.
   * :guilabel:`Manual`: In this mode, you have to remember to manually adjust the Strength knob to a desired level each time you speak. Otherwise, the music will play at the volume set by the Strength knob, regardless of the volume coming through the microphone input. When you speak through your microphone, set the volume to which the music should reduce by turning the Strength knob to a level that suits you.
```

>    * :guilabel:`Manual`: In this mode, you have to remember to manually adjust the Strength knob to a desired level each time you speak. Otherwise, the music will play at the volume set by the Strength knob, regardless of the volume coming through the microphone input. When you speak through your microphone, set the volume to which the music should reduce by turning the Strength knob to a level that suits you.

Reviewer: this is true, but I think the intended behavior is for the user to use the Talk button to control this.

"In this mode, the music will play at the volume set by the Strength knob whenever the microphone Talk button is active, regardless of the volume coming through the microphone input. When you speak through your microphone, set the volume to which the music should reduce by turning the Strength knob to a level that suits you. When you disable Talk mode, the music will return to the regular volume"

Author: Oh. Alright.

.. hint:: To set the Ducking volume correctly, you can use manual mode to adjust the knob to your taste and then set to auto when happy.

> .. Note:: Enabling the Microphone Gain option will cause the ducking effect to be ineffective, so, you can disable it for the ducking effect to work correctly.

Reviewer: @deborahtrez  where did this note come from? When you say "ineffective" do you mean "less effective", or "not effective at all?"

Author: When I enabled the Microphone Gain, and used microphone ducking, I noticed that the two would not work together (at all). Even with very low and very high ducking Strengths. The Gain option had to be off/disabled for the ducking to work.

Author: Is this note worthy?  Perhaps i should leave it out?

Reviewer: Yes that seems like a bug.  Leave it out and then this is LGTM

Reviewer 2: Please file a bug report on Launchpad then.

Reviewer 3: > The Gain option had to be off/disabled for the ducking to wor

Are you talking about the tiny Gain 'mute' button in Deere 2.2?
When the input is muted there's no reason for the Ducking to kick in, so I'd say this is intended behaviour.

Reviewer 3: Those gain buttons were removed for Mixxx 2.3 as they only cause confusion and we added PLAY buttons to Aux decks as equivalent for mics' TALK buttons for a consistent UI.

Author: I have filed the report [here](https://bugs.launchpad.net/mixxx/+bug/1903754) on the launchpad

Reviewer 3: Marked as Invalid because the source of confusion is that the GAIN label should describe the Gain knob only but is right next to the Mute button.
So you discovered a UX issue instead :)

Reviewer 3: Repairing the UI in 2.2 would be the right fix, but I doubt the effort is worth it since we're rather close to publishing 2.3.

Author: Okay. Let me remove this note section and commit.

Clicking  on the :guilabel:`Auto` button will switch it to any of three modes:

   * Off: When you speak through the microphone, the music  volume does not reduce.

>    * Auto: When you speak through the microphone, the music volume will be reduced when Mixxx detects your voice. You can adjus4t the amount by which the volume is automatically reduced with the Strength knob.

Reviewer 2: This introduces a typo.

Author: Ohhh my God. Let me get that fixed right now.

Author: hmmm....check still didn't pass..
